lib/elixir/lib/kernel/cli.ex

```diff
@@ -280,6 +289,16 @@ defmodule Kernel.CLI do
     {[], new_config}
   end
 
+  defp parse_shared(["--rpc-call", node, mod, fun | args], config) do
```

Comment:
You don't want to pass all args. For example, this should still work:

    --rpc-call node IO inspect --eval "IO.inspect(:local)" :remote

You also want this to work:

    --rpc-call node IO inspect -- :remote

The idea is to store only `node`, `mod`, and `fun` here. The args should be retrieved from `System.argv`.